lib/src/mopinion_platform.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/services.dart';
+import 'package:mopinion_flutter_integration_plugin/src/mopinion_form_state.dart';
+
+class MopinionPlatform {
+  const MopinionPlatform();
+
+  static const _platform = MethodChannel("MopinionFlutterBridge/native");
+  static const _events = EventChannel("MopinionFlutterBridge/native/events");
+
+  static const _eventAction = "trigger_event";
+  static const _initSdkAction = "init_sdk";
+  static const _addMetaDataAction = "add_meta_data";
+  static const _removeMetaDataAction = "remove_meta_data";
+  static const _removeAllMetaDataAction = "remove_all_meta_data";
+
+  Future<void> initSdk(
+    String deploymentKey, {
+    bool enableLogging = false,
+  }) {
+    try {
+      return _platform.invokeMethod(_initSdkAction, <String, dynamic>{
+        "deployment_key": deploymentKey,
+        "log": enableLogging
+      });
+    } on PlatformException catch (error, stackTrace) {
+      throw Exception(
+          'Initializing SDK failed: ${error.message}, stackTrace: $stackTrace');
+    }
+  }
+
+  Future<void> event(String eventName) {
+    try {
+      return _platform.invokeMethod(
+          _eventAction, <String, dynamic>{"argument1": eventName});
+    } on PlatformException catch (error, stackTrace) {
+      throw Exception(
+          'Launching event $eventName failed: ${error.message}, stackTrace: $stackTrace');
+    }
+  }
+
+  Stream<MopinionFormState> eventsData() => _events
+      .receiveBroadcastStream()
+      .map((event) => (event as String).toMopinionFormState());
+
+  Future<void> data(Map<String, String> map) async {
+    for (final entry in map.entries) {
+      try {
+        await _platform.invokeMethod(_addMetaDataAction,
+            <String, dynamic>{"key": entry.key, "value": entry.value});
+      } on PlatformException catch (error, stackTrace) {
+        throw Exception(
+            'Adding meta data for key ${entry.key} failed: ${error.message}, stackTrace: $stackTrace');
```

Comment:
I would have to look into the details of the native code, but would be nice if these exceptions can be made more specific based on the platform exception. For example `SocketException` if there is no internet connection and a `MopinionServerException` if the server is down.